### PR TITLE
fix custody limit always after fulu

### PR DESCRIPTION
**Pull request by tbenr**
Always consider custody no earlier than fulu activation

## Documentation

- [ ] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [ ] I thought about adding a changelog entry, and added one if I deemed necessary.

---

> [!NOTE]
> Clamp min custody period to FULU activation and remove redundant milestone gating; add unit tests for boundary cases.
> 
> - **Custody logic**:
>   - Update `MinCustodyPeriodSlotCalculator` to cap `minCustodyEpoch` at FULU activation (`max(fuluActivationEpoch)`), ensuring custody starts no earlier than FULU.
>   - Simplify FULU config access by requiring FULU (`orElseThrow`) and using `getMinEpochsForDataColumnSidecarsRequests`.
>   - In `DataColumnSidecarCustodyImpl`, remove explicit pre-FULU short-circuit; rely on calculator-based cutoff.
> - **Tests**:
>   - Add `MinCustodyPeriodSlotCalculatorTest` covering pre-/post-/at-activation boundaries.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 3873bf0e98438ae6f05a94f03a9a224717180813. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/datacolumns/MinCustodyPeriodSlotCalculator.java [hunk 1]
```diff
@@ -15,11 +15,14 @@
...
 import tech.pegasys.teku.infrastructure.unsigned.UInt64;
 import tech.pegasys.teku.spec.Spec;
+import tech.pegasys.teku.spec.SpecMilestone;
 import tech.pegasys.teku.spec.config.SpecConfigFulu;
 
 public interface MinCustodyPeriodSlotCalculator {
 
   static MinCustodyPeriodSlotCalculator createFromSpec(final Spec spec) {
+    final UInt64 fuluActivationEpoch =
+        spec.getForkSchedule().getFork(SpecMilestone.FULU).getEpoch();
```
[the remark this answers is not in the document]
Comment (tbenr) [reply ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/datacolumns/MinCustodyPeriodSlotCalculator.java:24]: I checked and we instantiate this class only after fulu support has been checked

**Review by zilm13 — DISMISSED** [review]
Looks good, but could we add any tests for it.
Just very simple tests for MinCustodyPeriodSlotCalculator without referring classes

**Comment by tbenr** [comment]
oky I'll add test, don't be rude 😉

**Review by zilm13 — APPROVED** [review]
LGTM
